### PR TITLE
Simplify `IRISPipeline` output.

### PR DESCRIPTION
**Pull request by wiktorlazarski**
# Simplify `IRISPipeline` output.

## PR description

Currently, `IRISPipeline` returns serialised version of generated `IrisTemplate` that causes confusions and make it less intuitive how one can use `HammingDistanceMatcher`. This PR simplifies `IRISPipeline` output such that together with `metadata`, unserialised `IrisTemplate` object is by default returned from `IRISPipeline` inference call. 

Also, update `ruff` tool setup to remove it's warnings and apply code quality tools to unify code base style etc.

### Issue

Without knowing more details about inner working of a code base, users find it difficult to simply take two IR images, generate `IrisTemplate`s and use `HammingDistanceMatcher` to get final distance between two irises.

### Solution

Introduce a new builder function that returns `IrisTemplate` object (if error wasn't raised during the `IRISPipeline` inference call) and `metadata` in a dictionary.

## Type

- [x] Feature
- [x] Refactoring

## Checklist

- [x] I've made sure that my code works as expected by writing unit tests.
- [x] I've checked if my code doesn't generate warnings or errors.
- [x] I've performed a self-review of my code.
- [x] I've made sure that my code follows the style guidelines of the project.
- [x] I've commented hard-to-understand parts of my code.
- [x] I've made appropriate changes in the documentation.

### CODE REVIEW

File: colab/GettingStarted.ipynb [hunk 1]
```diff
@@ -221,9 +223,9 @@
...
+    "Each code available in `output[\"iris_template\"]` object is a `numpy.ndarray` of shape `(16, 256, 2)`. The length of arrays containing iris codes and mask codes is determined by `IRISPipeline` filter bank parameters. The iris/mask code shape's dimmensions correspond to the following `(iris_code_height, iris_code_width, 2)`. Values `iris_code_height` and `iris_code_width` are determined by `ProbeSchema`s defined for `ConvFilterBank` object and `num_filters` is determined by number of filters specified for `ConvFilterBank` object. The last `2` value of the iris/mask code dimmension corresponds to real and complex parts of each complex filter response.\n",
```
Comment (ycbiometrics) [inline colab/GettingStarted.ipynb:228]: "to the real and imaginary parts of each complex filter response", "are determined by ProbeSchema's definition for ..."

File: docs/source/examples/getting_started.rst [hunk 2]
```diff
@@ -52,9 +52,9 @@ If ``output["error"]`` value is ``None``, ``IRISPipeline`` finished inference ca
...
+Each code available in ``output["iris_template"]`` dictionary is a ``numpy.ndarray`` of shape ``(16, 256, 2)``. The length of arrays containing iris codes and mask codes is determined by ``IRISPipeline`` filter bank parameters. The iris/mask code shape's dimensions correspond to the following ``(iris_code_height, iris_code_width, 2)``. Values ``iris_code_height`` and ``iris_code_width`` are determined by ``ProbeSchema``s defined for ``ConvFilterBank`` object and ``num_filters`` is determined by number of filters specified for ``ConvFilterBank`` object. The last ``2`` value of the iris/mask code dimension corresponds to real and complex parts of each complex filter response.
```
Comment (ycbiometrics) [inline docs/source/examples/getting_started.rst:57]: same here: "to the real and imaginary parts of each complex filter response", "are determined by ProbeSchema's definition for ..."

**Review by ycbiometrics — COMMENTED** [review]
Thanks for the change. I think any effort to try to simply the open-source while maintaining the functionality will be greatly appreciated. Just one thought, why do we keep all the orb-related code/environment in here? Users won't access our orb anyways. Wouldn't it be better to remove them?

File: src/iris/orchestration/output_builders.py [hunk 3]
```diff
@@ -6,6 +6,43 @@
...
+    exception = call_trace.get_error()
+    if exception is None:
+        iris_template = call_trace["encoder"]
+        error = None
+    elif isinstance(exception, Exception):
+        error = {
+            "error_type": type(exception).__name__,
+            "message": str(exception),
+            "traceback": "".join(traceback.format_tb(exception.__traceback__)),
+        }
```
Comment (TanguyJeanneau) [inline src/iris/orchestration/output_builders.py:35]: @wiktorlazarski This is already dealt with in  `__get_error`, we can simplify this function even more 🙂

**Review by TanguyJeanneau — CHANGES_REQUESTED** [review]
Thanks for the contribution ! One minor comment on simplifying more `build_simple_output`, then it will be all good to go 🙌

File: src/iris/orchestration/output_builders.py (hunk 3 above)
Comment (wiktorlazarski) [reply src/iris/orchestration/output_builders.py:35]: Nice! I completely missed that. Thanks for the hint. That'll allow us to simplify also `build_orb_output` function.

**Review by TanguyJeanneau — APPROVED** [review]
LGTM ! 🙌 🌲

**Pull request merged**